lib/move-sync/__tests__/move-sync.test.js

```diff
@@ -0,0 +1,411 @@
+
+    try {
+      fse.moveSync(src, dest, {overwrite: true})
+    } catch (err) {
+      assert.ifError(err)
+    }
+
+    const contents = fs.readFileSync(dest, 'utf8')
+    const expected = /^sonic the hedgehog\r?\n$/
+    assert.ok(contents.match(expected), `${contents} match ${expected}`)
+  })
+
+  it('should overwrite the destination directory if overwrite = true', function (done) {
+    // Tests fail on appveyor/Windows due to
+    // https://github.com/isaacs/node-graceful-fs/issues/98.
+    // Workaround by increasing the timeout by a minute (because
+    // graceful times out after a minute).
+    this.timeout(90000)
+
+    // Create src
+    const src = path.join(TEST_DIR, 'src')
+    fse.ensureDirSync(src)
+    fse.mkdirsSync(path.join(src, 'some-folder'))
+    fs.writeFileSync(path.join(src, 'some-file'), 'hi')
+
+    const dest = path.join(TEST_DIR, 'a-folder')
+
+    // verify dest has stuff in it
+    const pathsBefore = fs.readdirSync(dest)
+    assert(pathsBefore.indexOf('another-file') >= 0)
+    assert(pathsBefore.indexOf('another-folder') >= 0)
+
+    try {
+      fse.moveSync(src, dest, {overwrite: true})
+    } catch (err) {
+      assert.ifError(err)
+    }
+
+    // verify dest does not have old stuff
+    const pathsAfter = fs.readdirSync(dest)
+    assert.strictEqual(pathsAfter.indexOf('another-file'), -1)
+    assert.strictEqual(pathsAfter.indexOf('another-folder'), -1)
+
+    // verify dest has new stuff
+    assert(pathsAfter.indexOf('some-file') >= 0)
+    assert(pathsAfter.indexOf('some-folder') >= 0)
+    done()
+  })
+
+  /*
+  it('should not create directory structure if mkdirp is false', done => {
+    const src = `${TEST_DIR}/a-file`
+    const dest = `${TEST_DIR}/does/not/exist/a-file-dest`
+
+  // verify dest directory does not exist
+    assert(!fs.existsSync(path.dirname(dest)))
+
+    fse.move(src, dest, {mkdirp: false}, err => {
+      assert.strictEqual(err.code, 'ENOENT')
+      done()
+    })
+  })
+  */
+
+  it('should create directory structure by default', () => {
+    const src = `${TEST_DIR}/a-file`
+    const dest = `${TEST_DIR}/does/not/exist/a-file-dest`
+
+    // verify dest directory does not exist
+    assert(!fs.existsSync(path.dirname(dest)))
+
+    try {
+      fse.moveSync(src, dest)
+    } catch (err) {
+      assert.ifError(err)
+    }
+
+    const contents = fs.readFileSync(dest, 'utf8')
+    const expected = /^sonic the hedgehog\r?\n$/
+    assert.ok(contents.match(expected), `${contents} match ${expected}`)
+  })
+
+  it('should work across devices', () => {
+    const src = `${TEST_DIR}/a-file`
+    const dest = `${TEST_DIR}/a-file-dest`
+
+    setUpMockFs('EXDEV')
+
+    try {
+      fse.moveSync(src, dest)
+    } catch (err) {
+      assert.ifError(err)
+    }
+
+    const contents = fs.readFileSync(dest, 'utf8')
+    const expected = /^sonic the hedgehog\r?\n$/
+    assert.ok(contents.match(expected), `${contents} match ${expected}`)
+    tearDownMockFs('EXDEV')
```

Comment:
No need to pass a parameter here, `tearDownMockFs` doesn't use it. Correct other places where this is done.